actors/builtin/miner/miner_state.go

```diff
@@ -259,6 +272,129 @@ func (st *State) ComputeProvingSet(store adt.Store) ([]abi.SectorInfo, error) {
 	return sectorInfos, nil
 }
 
+func (st *State) addLockedFund(store adt.Store, currEpoch abi.ChainEpoch, pledgeAmount abi.TokenAmount) error {
+	Assert(pledgeAmount.GreaterThanEqual(big.Zero()))
+
+	lockedFundsArray := adt.AsArray(store, st.LockedFundsArray)
+
+	cliffStartEpoch := currEpoch + PledgeCliff
+	fullyVestedEpoch := currEpoch + PledgeCliff + PledgeVestingPeriod
+
+	for i := cliffStartEpoch; i < fullyVestedEpoch; i += VestIncrement {
```

Comment:
Quick order of magnitude for how many times this loop executes on average?

Reply:
ah, from the constants above it looks like 365 times. Thats gonna be reallly expensive. Can we cut this down to maybe like 10?

Reply:
Also, i think its extremely unlikely that you'll get any overlap here if youre keying this on a block height, and using the current epoch as your base. Maybe would be better if you rounded it to every ~10k blocks or something?

Reply:
Basically, as written, youre gonna rewrite the entire AMT every time this method is called. Thats not acceptable

Reply:
We can parametrize this to vest on a weekly basis and the quantization will also help.